napari/layers/image/image.py

```diff
@@ -729,105 +732,86 @@ def _raw_to_displayed(self, raw):
-                        1,
-                    )
-                self._transforms['tile2data'].translate = (
-                    self.corner_pixels[0] * self._transforms['tile2data'].scale
-                )
-            image = self.data[level][tuple(indices)]
-            image_indices = indices
-
-            # Slice thumbnail
-            indices = np.array(self._slice_indices)
-            downsampled_indices = (
-                indices[not_disp]
-                / self.downsample_factors[self._thumbnail_level, not_disp]
-            )
-            downsampled_indices = np.round(
-                downsampled_indices.astype(float)
-            ).astype(int)
-            downsampled_indices = np.clip(
-                downsampled_indices,
-                0,
-                self.level_shapes[self._thumbnail_level, not_disp] - 1,
-            )
-            indices[not_disp] = downsampled_indices
+        # For the old experimental async code.
+        self._empty = False
 
-            thumbnail_source = self.data[self._thumbnail_level][tuple(indices)]
-        else:
-            self._transforms['tile2data'].scale = np.ones(self.ndim)
-            image_indices = self._slice_indices
-            image = self.data[image_indices]
-
-            # For single-scale we don't request a separate thumbnail_source
-            # from the ChunkLoader because in ImageSlice.chunk_loaded we
-            # call request.thumbnail_source() and it knows to just use the
-            # image itself is there is no explicit thumbnail_source.
-            thumbnail_source = None
-
-        # Load our images, might be sync or async.
-        data = self._SliceDataClass(
-            self, image_indices, image, thumbnail_source
+        # The new slicing code makes a request from the existing state and
+        # executes the request on the calling thread directly.
+        # For async slicing, the calling thread will not be the main thread.
+        request = self._make_slice_request_internal(self._slice_input, indices)
+        response = request()
+        self._update_slice_response(response, indices)
+
+    def _make_slice_request(self, dims: Dims) -> _ImageSliceRequest:
+        """Make an image slice request based on the given dims and this image."""
+        slice_input = self._make_slice_input(
+            dims.point, dims.ndisplay, dims.order
+        )
+        # TODO: for the existing sync slicing, slice_indices is passed through
+        # to avoid some performance issues related to the evaluation of the
+        # data-to-world transform and its inverse. Async slicing currently
+        # absorbs these performance issues here, but we can likely improve
+        # things either by caching the world-to-data transform on the layer
+        # or by lazily evaluating it in the slice task itself.
+        slice_indices = slice_input.data_indices(self._data_to_world.inverse)
+        return self._make_slice_request_internal(slice_input, slice_indices)
```

Comment:
I'm a bit confused by breaking up `_make_slice_request` and `make_slice_requet_internal`. Why not just include these in a single method?

Reply:
`_ImageBase._set_view_slice` doesn't have access to `Dims` so can't call `_make_slice_request`. I describe this as temporary scaffolding that should go away once `_set_view_slice` does.

Alternatively, we could change the expected signature of `_make_slice_request` to accommodate the signature of `_slice_dims` (i.e. by passing through only some attributes of `Dims`), but I liked the conciseness of just passing the argument through there.